### ISSUE TITLE
Sudden influx of "SMTP Error: Could not authenticate." starting March 30, 2021

### ISSUE THREAD

**Issue by perelx** [PHPMailer/PHPMailer#2314]
# Problem description
Nothing had changed on my end. Starting promptly on March 30, 2021, several of my servers constantly receive an error of:

`SMTP Error: Could not authenticate.`

To be clear: emails are still sending successfully. I've since set a flag in my code to try to resend the email when this happens. They all successfully resend. As far as I can tell, I didn't change anything to cause this. And since this happened across several servers, not related to each other, I'm assuming it's a higher level issue.

My environment:

- Mix of Ubuntu 18 and 20 stable editions, all servers updated
- Mix of PHP 7.4 and 8.0, latest updates
- PHPMailer 6.4 (latest stable)
- AWS, using SES credentials
- AWS SES dedicated IP, with a very high rate /per second email send, and 500K/24 hours. I don't come anywhere near close to my limits

Again, this seemed to appear all at the same time, and across multiple servers. I keep a table log of emails that generate errors, and my logs have been clear for years, minus some edge exceptions. All of a sudden, 2021-03-30, my table is filled daily with the same error message. Most emails send. And any emails that have an auth error are marked to be resent the next minute, and they send successfully on the next go around.

I'm hoping someone has run into something similar.

# Debug output
SMTP Error: Could not authenticate.

**Comment by Synchro** [comment]
Without detailed debug info, there's not really anything I can do. Enable debug output (`SMTPDebug = 3`) and write it to a log (inject a closure into the `debugoutput` property to do that). PHPMailer's authentication mechanisms have not changed recently, and I have little idea about AWS. It could be something silly like one of the many SES servers having an expired certificate that you hit often but randomly. Debug info will tell you more.

**Comment by perelx** [comment]
> Without detailed debug info, there's not really anything I can do. Enable debug output (`SMTPDebug = 3`) and write it to a log (inject a closure into the `debugoutput` property to do that). PHPMailer's authentication mechanisms have not changed recently, and I have little idea about AWS. It could be something silly like one of the many SES servers having an expired certificate that you hit often but randomly, or a problem with daylight savings changes. Debug info will tell you more.

Thank you, I'll see if I can get the proper debug info and update this

**Comment by perelx** [comment]
@Synchro 

I figured it out. It was, indeed, AWS. It's a signature version issue. We are supposed to be using their version 4 (version 2 is deprecated). For anyone else:

I cycled all of my SES IAM credentials. But it looks like I did not do the correct steps to upgrade to version 4. I was supposed to create an entirely new IAM user via the SES dashboard (I use the web dashboard). This then upgraded the signature.

As I wrote this, the server using the upgraded IAM user sent ~60 emails with no SMTP Auth errors. This had to be the issue.
